io.asgardeo.tomcat.saml.agent/src/main/java/io/asgardeo/tomcat/saml/agent/SAML2SSOAgentFilter.java

```diff
@@ -86,76 +86,122 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
 
             if (resolver.isSLORequest()) {
 
-                samlSSOManager = new SAML2SSOManager(ssoAgentConfig);
-                LogoutResponse logoutResponse = samlSSOManager.doSLO(request);
-                String encodedRequestMessage = samlSSOManager.buildPostResponse(logoutResponse);
-                SSOAgentUtils.sendPostResponse(request, response, encodedRequestMessage);
+                try {
+                    samlSSOManager = new SAML2SSOManager(ssoAgentConfig);
+                    LogoutResponse logoutResponse = samlSSOManager.doSLO(request);
+                    String encodedRequestMessage = samlSSOManager.buildPostResponse(logoutResponse);
+                    SSOAgentUtils.sendPostResponse(request, response, encodedRequestMessage);
+                } catch (SSOAgentException e) {
+                    handleException(request, response, ssoAgentConfig, e);
+                    return;
+                }
                 return;
             } else if (resolver.isSAML2SSOResponse()) {
 
-                samlSSOManager = new SAML2SSOManager(ssoAgentConfig);
                 try {
+                    samlSSOManager = new SAML2SSOManager(ssoAgentConfig);
                     samlSSOManager.processResponse(request, response);
+                    if (!"true".equals(request.getAttribute(SSOAgentConstants.SHOULD_GO_TO_WELCOME_PAGE))) {
```

Comment:
Shall we use Parse boolean here?

Reply:
Addressed with https://github.com/asgardeo/asgardeo-tomcat-saml-agent/pull/47/commits/dee9d739015733c812fbefca25ffab28bc6b2ffd